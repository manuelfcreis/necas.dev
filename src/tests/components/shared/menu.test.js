import React from "react";
import menuWithRouter, { Menu } from "../../../components/shared/menu";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow } from "enzyme";

it("renders correctly", () => {
  const element = shallow(
    <Router>
      <menuWithRouter
        positions={{ x: 50, y: 50 }}
        location={{ pathname: "/" }}
      />
    </Router>
  );
  expect(element).toMatchSnapshot();
});

it("renders correctly", () => {
  const element = shallow(
    <Router>
      <menuWithRouter
        positions={{ x: 50, y: 50 }}
        location={{ pathname: "about" }}
      />
    </Router>
  );
  expect(element).toMatchSnapshot();
});

it("renders correctly", () => {
  const element = shallow(
    <Router>
      <menuWithRouter
        positions={{ x: 50, y: 50 }}
        location={{ pathname: "blog" }}
      />
    </Router>
  );
  expect(element).toMatchSnapshot();
});

it("renders correctly", () => {
  const element = shallow(
    <Menu positions={{ x: 50, y: 50 }} location={{ pathname: "about" }} />
  );
  expect(element).toMatchSnapshot();
});
