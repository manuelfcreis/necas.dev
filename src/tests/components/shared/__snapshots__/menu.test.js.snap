// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <menuWithRouter
    location={
      Object {
        "pathname": "/",
      }
    }
    positions={
      Object {
        "x": 50,
        "y": 50,
      }
    }
  />
</Router>
`;

exports[`renders correctly 2`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <menuWithRouter
    location={
      Object {
        "pathname": "about",
      }
    }
    positions={
      Object {
        "x": 50,
        "y": 50,
      }
    }
  />
</Router>
`;

exports[`renders correctly 3`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <menuWithRouter
    location={
      Object {
        "pathname": "blog",
      }
    }
    positions={
      Object {
        "x": 50,
        "y": 50,
      }
    }
  />
</Router>
`;

exports[`renders correctly 4`] = `
<div
  className="menu"
>
  <ul>
    <Eye
      hide={false}
      positions={
        Object {
          "x": 50,
          "y": 50,
        }
      }
      small={true}
    />
    <li>
      <Link
        to="/about"
      >
        About me
      </Link>
    </li>
    <li>
      <Link
        to="/blog"
      >
        Blog
      </Link>
    </li>
    <li>
      <a
        href="https://www.dropbox.com/s/9vlsiv339xoipzo/resume.pdf?dl=0"
        rel="noopener noreferrer"
        target="_blank"
      >
        Resume
      </a>
    </li>
    <li>
      <a
        href="https://github.com/manuelfcreis"
        rel="noopener noreferrer"
        target="_blank"
      >
        Github
      </a>
    </li>
  </ul>
</div>
`;
